### ISSUE TITLE
Icon "placeholder" sometimes showing when pointing to an app

### ISSUE THREAD

**Issue by BigMacAdmin** [swiftDialog/swiftDialog#235]
![image](https://user-images.githubusercontent.com/106293503/220525270-2221594d-e031-4d17-af27-7cb4ca542c58.png)
![image](https://user-images.githubusercontent.com/106293503/220525446-b47ad1e6-10b2-45c0-a3dc-e9504781a5b9.png)

The exact same script on the same VM was used to generate both of these screenshots, however one time the icon drew and one time it didn't.

It seems to happen consistently the very first time I run the script on a device/VM, and only happens when using a `/foo/bar/Something.app` path (not a file path to an .icns or .png). Subsequent runs will load the .app icon properly.

This is the output (via `set -x`) from the failed run

```
2023-02-21 20:47:16.927 Dialog[2378:29496] XType: failed to connect - Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process." UserInfo={NSDebugDescription=The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process.}
2023-02-21 20:47:16.927 Dialog[2378:29496] Font server protocol version mismatch (expected:5 got:0), falling back to local fonts
2023-02-21 20:47:16.927 Dialog[2378:29496] XType: unable to make a connection to the font daemon!
2023-02-21 20:47:16.927 Dialog[2378:29496] XType: XTFontStaticRegistry is enabled as fontd is not available.
```
This is the output (via `set -x`) from the successful run
```
2023-02-21 21:00:25.215 Dialog[1494:10410] XType: failed to connect - Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process." UserInfo={NSDebugDescription=The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process.}
2023-02-21 21:00:25.215 Dialog[1494:10410] Font server protocol version mismatch (expected:5 got:0), falling back to local fonts
2023-02-21 21:00:25.215 Dialog[1494:10410] XType: unable to make a connection to the font daemon!
2023-02-21 21:00:25.215 Dialog[1494:10410] XType: XTFontStaticRegistry is enabled as fontd is not available.
```
I think the font errors are a red herring, i see those often when running SwiftDialog from a management tool or from a LaunchDaemon. In this case, the script is running from a LaunchDaemon.

This is the dialog command in question. You can see its calling a local macOS system app for the icon:
`+/usr/local/Baseline/Baseline.sh:903> /usr/local/bin/dialog --title 'Your computer setup is underway' --message 'Feel free to step away, this could take 30 minutes or more. \n\nYour computer will restart when it'\''s ready for use.' --icon '/System/Library/CoreServices/Erase Assistant.app' --blurscreen --width 900 --height 550 --button1disabled --commandfile /var/tmp/baselineDialog.4cIrR8 --quitkey ']' --listitem 'Google Chrome' --listitem Firefox --listitem 'Omnidisk Sweeper' --listitem Zoom --listitem Dockutil --listitem 'Microsoft Office' --listitem Nextiva --listitem 'SSC Customization' --listitem 'User Settings'`

**Comment by mm2270** [comment]
I've had this issue happen to me too. In most of my cases, the icon initially shows up as the generic talk bubble icon that swiftDialog uses when no icon is specified or found, and then switches to the actual icon. The time it takes to make that switch is usually a couple of seconds. I've had a few cases where the icon doesn't appear at all, like above.

It seems the application has some trouble initially determining the icon from an app bundle and falls back to its defaults.

**Comment by joshua-d-miller** [comment]
I am also seeing this issue particularly with App bundles when running an initial onboarding dialog. Also I have one of the icons set as a SF Icon from SF Symbols and it does not display in the list but displays when in the bigger side of a progress dialog.

**Comment by mm2270** [comment]
I'm using Dialog version 2.1.0.4148, and I keep running into this problem. It seems very consistent now that app icons from bundles don't appear correctly in the resulting dialog. I'm uncertain if the problem is when the script is called as root from a Jamf policy. This is often when I see it. If I just call Dialog directly from the shell and use an app bundle path for the icon, it appears just fine. It also seems to happen when using lists very consistently, but not necessarily with a standard dialog.

Is this being looked into? It's not causing the dialogs to fail, but it's messing up the dialogs from an aesthetic point of view, so it would be great to see this fixed at some point.

**Comment by bartreardon** [comment]
@mm2270 if you could provide an example where it happens consistently for you that would help. in my own testing here I can't replicate what you are seeing.

**Comment by joncrain** [comment]
I've been seeing this as well with `"icon": "/System/Library/CoreServices/Installer.app"`
 
In our case it's running as root as part of our bootstrap process. Sometimes its there, sometimes its not 🤷

**Comment by BigMacAdmin** [comment]
Other conversation about it on Slack as well:
https://macadmins.slack.com/archives/C01U5MXNGG6/p1679681658584889

**Comment by bartreardon** [comment]
as per [this comment](https://github.com/aonez/Keka/issues/1177#issuecomment-1539820458) and also [this post](https://forums.macrumors.com/threads/when-does-the-icon-cache-clear-itself.2368253/post-32152155) this is an identified bug in Ventura and looks like it will be fixed in macOS 14.

**Comment by bartreardon** [comment]
interested if this is fixed in [2.3](https://github.com/bartreardon/swiftDialog/releases/tag/v2.3.0)

**Comment by bartreardon** [comment]
should be fixed in 2.3.1 - can re-open if it's still a problem
